src/xma/src/xmaapi/xmaxclbin.cpp

```diff
@@ -64,284 +59,33 @@ std::vector<char> xma_xclbin_file_open(const std::string& xclbin_name)
-        }
-        xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "Num of soft kernels on this device = %ud ", num_soft_kernels);
-
-        xclbin_info->number_of_kernels = xma_ip_layout.size();
-        xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "Num of total kernels on this device = %ud ", xclbin_info->number_of_kernels);
-
-        xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "  ");
-        const axlf_section_header *xml_hdr = xclbin::get_axlf_section(xclbin, EMBEDDED_METADATA);
-        if (xml_hdr) {
-            char *xml_data = const_cast<char*>(&buffer[xml_hdr->m_sectionOffset]);
-            uint64_t xml_size = xml_hdr->m_sectionSize;
-            if (xml_size > 0 && xml_size < 500000) {
-                xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "XML MetaData is:");
-                xma_core::utils::streambuf xml_streambuf(xml_data, xml_size);
-                std::istream xml_stream(&xml_streambuf);
-                std::string line;
-                while(std::getline(xml_stream, line)) {
-                    xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "%s", line.c_str());
-                }
-            }
-        } else {
-            xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "XML MetaData is missing");
-        }
-        xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "  ");
-        const axlf_section_header *kv_hdr = xclbin::get_axlf_section(xclbin, KEYVALUE_METADATA);
-        if (kv_hdr) {
-            char *kv_data = const_cast<char*>(&buffer[kv_hdr->m_sectionOffset]);
-            uint64_t kv_size = kv_hdr->m_sectionSize;
-            if (kv_size > 0 && kv_size < 200000) {
-                xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "Key-Value MetaData is:");
-                xma_core::utils::streambuf kv_streambuf(kv_data, kv_size);
-                std::istream kv_stream(&kv_streambuf);
-                std::string line;
-                while(std::getline(kv_stream, line)) {
-                    uint32_t lsize = line.size();
-                    uint32_t pos = 0;
-                    while (pos < lsize) {
-                        xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "%s", line.substr(pos, MAX_KERNEL_NAME).c_str());
-                        pos += MAX_KERNEL_NAME;
-                    }
-                }
-            }
-        } else {
-            xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "Key-Value Data is not present in xclbin");
-        }
-        xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD, "  ");
-    }
-    else
-    {
+    const axlf* xclbin_ax = reinterpret_cast<const axlf*>(buffer);
+    const axlf_section_header* ip_hdr = xclbin::get_axlf_section(xclbin_ax, IP_LAYOUT);
+    if (!ip_hdr) {
         xma_logmsg(XMA_ERROR_LOG, XMAAPI_MOD, "Could not find IP_LAYOUT in xclbin ip_hdr=%p ", ip_hdr);
-        throw std::runtime_error("Could not find IP_LAYOUT in xclbin");
+        throw std::runtime_error("Could not find IP_LAYOUT in xclbin file");
     }
+    const char* data = &buffer[ip_hdr->m_sectionOffset];
+    info->ip_axlf = reinterpret_cast<const ip_layout*>(data);
```

Comment:
I wonder if this function should create the xrt:;xclbin object and store it in  `info` for continued use?

Reply:
addressed this comment and used XRT::XCLBIN class in the code.